core/commands/dag/dag.go

```diff
@@ -303,7 +303,11 @@ pin each individual root specified in the car headers, before GC runs again.
 		cmds.BoolOption(pinRootsOptionName, "Pin optional roots listed in the .car headers after importing.").WithDefault(true),
 	},
 	PreRun: func(req *cmds.Request, env cmds.Environment) error {
-		if silent, _ := req.Options[silentOptionName].(bool); silent {
+		silent, _ := req.Options[silentOptionName].(bool)
+		encType, _ := req.Options[cmds.EncLong].(string)
+
+		// force-disable progress unless on non-silent CLI
+		if silent || encType != "text" {
```

Comment:
@Stebalien is this reasonable to do here, or do we have a centralized way to switching on this? I suppose also having TTY detection would be nice somewhere centrally...

Reply:
If we're going to check this, we should do so inside `PostRun`.

Reply:
@Stebalien there are 2 things here, to clarify:
 
- Checking/resetting stuff coming from req.Options is ok in PreRun
- Checking for whether stderr is a tty cn only happen in PostRun

Correct?

Reply:
> Checking/resetting stuff coming from req.Options is ok in PreRun

In general, yes. You can do this if you need to change some flag before it's sent to the server.

> Checking for whether stderr is a tty cn only happen in PostRun

Yes. PreRun and PostRun are always run on the local terminal. Run may run on the local terminal or may run on a remote terminal.

Reply:
@Stebalien generalized as https://github.com/ipfs/go-ipfs/pull/7011/commits/de52029e1894de636f9c6953503111ffac6493ef